### PR TITLE
do not center text when not applicable

### LINKED ISSUE #1513: Inserting newline and text in a previously existing text object: another newline gets added above
### How to reproduce
1. start with fresh canvas
2. enable the text tool
3. click in the empty space to insert text
4. type a single letter (let's say an "m"), then Enter, then another m
5. click somewhere else to end the text entering/loose focus from the text object
6. enable the text tool again
7. click on the text object
8. press Ctrl+Enter to go to the very end
9. press Enter, type another m
10. click somewhere else to end the text entering/loose focus from the text object

### What happens
After step 10, when the text object looses focus, a newline appears between the first and the second line. Also, the text object moves to the top.

### What should happen
The y position of the first line should be the same/the text object should not move to the top. There should not be another newline appearing between line 1 and 2.

### Environment
Ubuntu 19.10, Firefox 75.0

After step 4 | After step 8 | After step 10
------------ | ------------- | -------------
Screenshot from 2020-04-29 20-10-25 | Screenshot from 2020-04-29 20-10-46 | Screenshot from 2020-04-29 20-10-56

When I pressed “Print Screen” on my keyboard after step 4, this had the side effect to select the text object (finishing editing). The same happened when I pressed “Print Screen” after step 9; and this had the same effect as step 10. (I carefully cropped the screenshots exactly the same).

This is a “minimal example”. The same effect (newlines appearing in the middle of a text) with larger texts; there, newlines would appear at multiple locations, basically doubling empty lines.

### PR DESCRIPTION
**Pull request by dwelle**
- fixes https://github.com/excalidraw/excalidraw/issues/1327, https://github.com/excalidraw/excalidraw/issues/1326
- migrating to textarea fixes #1513, #1512, #1809
- (un)related bugfix #1792, #1226 (I think)

---

- Text is no longer centered vertically/horizontally when typing. **Except** for when you create text at element center, where I kept the centering behavior as previously.
- Even though I introduced `element.verticalAlign`, I didn't expose this setting to users yet. Not sure if we even want to complicate Excalidraw in this way. But it's necessary to keep track of it internally.
- I named the prop `verticalAlign` to align with CSS, and similarly its values (`"top" | "middle"`) are aligned with CSS too. Maybe not a good idea, but it also helps to prevent typos (using `textAlign` in place of `verticalAlign`).
- Removed wysiwyg editor padding. Not sure what purpose it served, but it doesn't seem to serve any at this point.
- Migrated to `textarea` to normalize newlines across browsers so that the bounding box calculation is deterministic and aligned to the underlying text element being edited (for positioning reasons).

### CODE REVIEW

**Comment by kyldvs** [comment]
Just wanted to pop in and say awesome work :) This fixes one of my major pain points, excited to see it fixed!

**Comment by vjeux** [comment]
This is awesome!

One issue: Multi-line wysiwyg position is very different than canvas selection when zoomed in.

**Comment by lipis** [comment]
Yes, it's jumping a bit when zoomed in.

**Comment by dwelle** [comment]
Good catch!

**Comment by dwelle** [comment]
Made some significant changes to fix the bug in OP; may have regressed previously working behavior.

**Blocking realease:** Wysiwyg is completely broken around rotated text. Not sure how I'm gonna fix it, but will leave it for tomorrow+.

File: src/math.ts [hunk 1]
```diff
@@ -57,57 +57,48 @@ export const rotate = (
   ];
 
 export const adjustXYWithRotation = (
-  side: "n" | "s" | "w" | "e" | "nw" | "ne" | "sw" | "se",
+  sides: {
+    n?: boolean;
+    e?: boolean;
+    s?: boolean;
+    w?: boolean;
+  },
```
Comment (dai-shi) [inline src/math.ts:65]: Very nice. I like it. Thanks!

File: src/element/resizeElements.ts [hunk 2]
```diff
@@ -235,6 +235,26 @@ const measureFontSizeFromWH = (
   return null;
 };
 
+const getSidesForResizeHandle = (
+  resizeHandle: "n" | "s" | "w" | "e" | "nw" | "ne" | "sw" | "se",
+  isResizeFromCenter: boolean,
+) => {
+  return {
+    n:
+      /^(n|ne|nw)$/.test(resizeHandle) ||
```
Comment (dai-shi) [inline src/element/resizeElements.ts:244]: If we really want, we could do this:
```suggestion
      /n/.test(resizeHandle) ||
```

File: src/element/newElement.ts [hunk 3]
```diff
@@ -171,34 +171,20 @@ const getAdjustedDimensions = (
...
-        deltaY2,
-        false,
-      );
-    }
+    [x, y] = adjustXYWithRotation(
+      {
+        s: true,
+        e: textAlign === "center" || textAlign === "left",
+        w: textAlign === "center" || textAlign === "right",
+      },
```
Comment (dai-shi) [inline src/element/newElement.ts:179]: 🎉 🎉 🎉

File: src/element/resizeElements.ts (hunk 2 above)
Comment (dwelle) [reply src/element/resizeElements.ts:244]: yea, or `/^n[ew]?$/` if we wanted to be safe.. dunno

**Comment by vjeux** [comment]
omg omg omg omg omg!

**Comment by getvega** [comment]
You rock thank you!!

**Pull request merged**